Test failure in CI build 2825

The following test appears to have failed:

[#2825](https://circleci.com/gh/cockroachdb/cockroach/2825):

```
I0524 01:27:56.993550     111 testserver.go:43] Test server listening on https: 127.0.0.1:43375
I0524 01:27:57.047406     111 client.go:150] client 127.0.0.1:43375 connected
I0524 01:27:57.047566     111 client.go:224] client 127.0.0.1:43375 starting heartbeat
W0524 01:27:57.049838     111 dist_sender.go:590] failed to invoke Scan: store.go:137: end key "a" must be greater than start "a"
I0524 01:27:57.056038     111 kv.go:112] failed Scan: store.go:137: end key "a" must be greater than start "a"
--- FAIL: TestStartEqualsEndKeyScan (0.31s)
	dist_sender_server_test.go:188: unexpected error on scan: store.go:137: end key "a" must be greater than start "a"
=== RUN TestMethods
I0524 01:27:57.297719     111 rocksdb.go:88] opening in-memory rocksdb instance
I0524 01:27:57.299556     111 range_command.go:764] range 1: new leader lease 0.000000000,0-1432430878.298724651,0 @1:1
I0524 01:27:57.300877     111 node.go:282] initialized store store=1:1 ([]=): {Capacity:257668677632 Available:199568736256 RangeCount:0 XXX_unrecognized:[]}
I0524 01:27:57.300973     111 gossip.go:165] gossiping node descriptor node_id:"\001" address:<network:"tcp" address:"127.0.0.1:56558" > attrs:<> 
I0524 01:27:57.301020     111 node.go:373] connecting to gossip network to verify cluster ID...
I0524 01:27:57.301187     111 range.go:929] gossiping first range from store 1, range 1
I0524 01:27:57.301230     111 range.go:937] gossiping sentinel from store 1, range 1
I0524 01:27:57.301388     111 range.go:988] gossiping accounting config from store 1, range 1
--
I0524 01:28:03.038822     111 server.go:177] starting https server at 127.0.0.1:34615
I0524 01:28:03.038883     111 testserver.go:43] Test server listening on https: 127.0.0.1:34615
I0524 01:28:03.102712     111 client.go:150] client 127.0.0.1:34615 connected
I0524 01:28:03.102883     111 client.go:224] client 127.0.0.1:34615 starting heartbeat
--- PASS: TestKeysAndBodyArePreserved (0.33s)
FAIL
FAIL	github.com/cockroachdb/cockroach/kv	15.705s
=== RUN TestHeartbeatSingleGroup
--- PASS: TestHeartbeatSingleGroup (0.09s)
=== RUN TestHeartbeatMultipleGroupsJointLeader
--- PASS: TestHeartbeatMultipleGroupsJointLeader (0.01s)
=== RUN TestHeartbeatResponseFanout
I0524 01:27:44.746267      94 heartbeat_test.go:387] SubmitCommand succeed after Heartbeat Response fanout
E0524 01:27:44.746653      94 transport.go:202] sending rpc failed: read tcp 127.0.0.1:52991: use of closed network connection
E0524 01:27:44.746711      94 transport.go:202] sending rpc failed: read tcp 127.0.0.1:52991: use of closed network connection
E0524 01:27:44.746760      94 transport.go:202] sending rpc failed: read tcp 127.0.0.1:52991: use of closed network connection
E0524 01:27:44.746811      94 transport.go:202] sending rpc failed: read tcp 127.0.0.1:58890: use of closed network connection
I0524 01:31:04.597529     305 testserver.go:43] Test server listening on https: 127.0.0.1:39865
I0524 01:31:05.059157     305 client.go:150] client 127.0.0.1:39865 connected
I0524 01:31:05.059685     305 client.go:224] client 127.0.0.1:39865 starting heartbeat
W0524 01:31:05.067198     305 dist_sender.go:590] failed to invoke Scan: store.go:137: end key "a" must be greater than start "a"
I0524 01:31:05.068027     305 kv.go:112] failed Scan: store.go:137: end key "a" must be greater than start "a"
--- FAIL: TestStartEqualsEndKeyScan (1.09s)
	dist_sender_server_test.go:188: unexpected error on scan: store.go:137: end key "a" must be greater than start "a"
=== RUN TestMethods
I0524 01:31:05.448861     305 rocksdb.go:88] opening in-memory rocksdb instance
I0524 01:31:05.453278     305 range_command.go:764] range 1: new leader lease 0.000000000,0-1432431066.450669289,0 @1:1
I0524 01:31:05.458009     305 node.go:282] initialized store store=1:1 ([]=): {Capacity:257668677632 Available:200138846208 RangeCount:0 XXX_unrecognized:[]}
I0524 01:31:05.458192     305 gossip.go:165] gossiping node descriptor node_id:"\001" address:<network:"tcp" address:"127.0.0.1:36774" > attrs:<> 
I0524 01:31:05.458401     305 node.go:373] connecting to gossip network to verify cluster ID...
I0524 01:31:05.459246     305 range.go:929] gossiping first range from store 1, range 1
I0524 01:31:05.459562     305 range.go:937] gossiping sentinel from store 1, range 1
I0524 01:31:05.460128     305 range.go:988] gossiping accounting config from store 1, range 1
--
I0524 01:31:43.568874     305 server.go:177] starting https server at 127.0.0.1:47606
I0524 01:31:43.569428     305 testserver.go:43] Test server listening on https: 127.0.0.1:47606
I0524 01:31:44.070755     305 client.go:150] client 127.0.0.1:47606 connected
I0524 01:31:44.071513     305 client.go:224] client 127.0.0.1:47606 starting heartbeat
--- PASS: TestKeysAndBodyArePreserved (1.40s)
FAIL
FAIL	github.com/cockroachdb/cockroach/kv	76.160s
=== RUN TestHeartbeatSingleGroup
--- PASS: TestHeartbeatSingleGroup (0.36s)
=== RUN TestHeartbeatMultipleGroupsJointLeader
--- PASS: TestHeartbeatMultipleGroupsJointLeader (0.02s)
=== RUN TestHeartbeatResponseFanout
I0524 01:30:44.885143     315 heartbeat_test.go:387] SubmitCommand succeed after Heartbeat Response fanout
E0524 01:30:44.886669     315 transport.go:202] sending rpc failed: read tcp 127.0.0.1:60816: use of closed network connection
E0524 01:30:44.886960     315 transport.go:202] sending rpc failed: read tcp 127.0.0.1:60816: use of closed network connection
E0524 01:30:44.887232     315 transport.go:202] sending rpc failed: read tcp 127.0.0.1:60816: use of closed network connection
E0524 01:30:44.887645     315 transport.go:202] sending rpc failed: read tcp 127.0.0.1:49835: use of closed network connection
I0524 01:31:04.597529     305 testserver.go:43] Test server listening on https: 127.0.0.1:39865
I0524 01:31:05.059157     305 client.go:150] client 127.0.0.1:39865 connected
I0524 01:31:05.059685     305 client.go:224] client 127.0.0.1:39865 starting heartbeat
W0524 01:31:05.067198     305 dist_sender.go:590] failed to invoke Scan: store.go:137: end key "a" must be greater than start "a"
I0524 01:31:05.068027     305 kv.go:112] failed Scan: store.go:137: end key "a" must be greater than start "a"
--- FAIL: TestStartEqualsEndKeyScan (1.09s)
	dist_sender_server_test.go:188: unexpected error on scan: store.go:137: end key "a" must be greater than start "a"
=== RUN TestMethods
I0524 01:31:05.448861     305 rocksdb.go:88] opening in-memory rocksdb instance
I0524 01:31:05.453278     305 range_command.go:764] range 1: new leader lease 0.000000000,0-1432431066.450669289,0 @1:1
I0524 01:31:05.458009     305 node.go:282] initialized store store=1:1 ([]=): {Capacity:257668677632 Available:200138846208 RangeCount:0 XXX_unrecognized:[]}
I0524 01:31:05.458192     305 gossip.go:165] gossiping node descriptor node_id:"\001" address:<network:"tcp" address:"127.0.0.1:36774" > attrs:<> 
I0524 01:31:05.458401     305 node.go:373] connecting to gossip network to verify cluster ID...
I0524 01:31:05.459246     305 range.go:929] gossiping first range from store 1, range 1
I0524 01:31:05.459562     305 range.go:937] gossiping sentinel from store 1, range 1
I0524 01:31:05.460128     305 range.go:988] gossiping accounting config from store 1, range 1
--
I0524 01:31:43.568874     305 server.go:177] starting https server at 127.0.0.1:47606
I0524 01:31:43.569428     305 testserver.go:43] Test server listening on https: 127.0.0.1:47606
I0524 01:31:44.070755     305 client.go:150] client 127.0.0.1:47606 connected
I0524 01:31:44.071513     305 client.go:224] client 127.0.0.1:47606 starting heartbeat
--- PASS: TestKeysAndBodyArePreserved (1.40s)
FAIL
FAIL	github.com/cockroachdb/cockroach/kv	76.160s
=== RUN TestHeartbeatSingleGroup
--- PASS: TestHeartbeatSingleGroup (0.36s)
=== RUN TestHeartbeatMultipleGroupsJointLeader
--- PASS: TestHeartbeatMultipleGroupsJointLeader (0.02s)
=== RUN TestHeartbeatResponseFanout
I0524 01:30:44.885143     315 heartbeat_test.go:387] SubmitCommand succeed after Heartbeat Response fanout
E0524 01:30:44.886669     315 transport.go:202] sending rpc failed: read tcp 127.0.0.1:60816: use of closed network connection
E0524 01:30:44.886960     315 transport.go:202] sending rpc failed: read tcp 127.0.0.1:60816: use of closed network connection
E0524 01:30:44.887232     315 transport.go:202] sending rpc failed: read tcp 127.0.0.1:60816: use of closed network connection
E0524 01:30:44.887645     315 transport.go:202] sending rpc failed: read tcp 127.0.0.1:49835: use of closed network connection
I0524 01:27:56.993550     111 testserver.go:43] Test server listening on https: 127.0.0.1:43375
I0524 01:27:57.047406     111 client.go:150] client 127.0.0.1:43375 connected
I0524 01:27:57.047566     111 client.go:224] client 127.0.0.1:43375 starting heartbeat
W0524 01:27:57.049838     111 dist_sender.go:590] failed to invoke Scan: store.go:137: end key "a" must be greater than start "a"
I0524 01:27:57.056038     111 kv.go:112] failed Scan: store.go:137: end key "a" must be greater than start "a"
--- FAIL: TestStartEqualsEndKeyScan (0.31s)
	dist_sender_server_test.go:188: unexpected error on scan: store.go:137: end key "a" must be greater than start "a"
=== RUN TestMethods
I0524 01:27:57.297719     111 rocksdb.go:88] opening in-memory rocksdb instance
I0524 01:27:57.299556     111 range_command.go:764] range 1: new leader lease 0.000000000,0-1432430878.298724651,0 @1:1
I0524 01:27:57.300877     111 node.go:282] initialized store store=1:1 ([]=): {Capacity:257668677632 Available:199568736256 RangeCount:0 XXX_unrecognized:[]}
I0524 01:27:57.300973     111 gossip.go:165] gossiping node descriptor node_id:"\001" address:<network:"tcp" address:"127.0.0.1:56558" > attrs:<> 
I0524 01:27:57.301020     111 node.go:373] connecting to gossip network to verify cluster ID...
I0524 01:27:57.301187     111 range.go:929] gossiping first range from store 1, range 1
I0524 01:27:57.301230     111 range.go:937] gossiping sentinel from store 1, range 1
I0524 01:27:57.301388     111 range.go:988] gossiping accounting config from store 1, range 1
--
I0524 01:28:03.038822     111 server.go:177] starting https server at 127.0.0.1:34615
I0524 01:28:03.038883     111 testserver.go:43] Test server listening on https: 127.0.0.1:34615
I0524 01:28:03.102712     111 client.go:150] client 127.0.0.1:34615 connected
I0524 01:28:03.102883     111 client.go:224] client 127.0.0.1:34615 starting heartbeat
--- PASS: TestKeysAndBodyArePreserved (0.33s)
FAIL
FAIL	github.com/cockroachdb/cockroach/kv	15.705s
=== RUN TestHeartbeatSingleGroup
--- PASS: TestHeartbeatSingleGroup (0.09s)
=== RUN TestHeartbeatMultipleGroupsJointLeader
--- PASS: TestHeartbeatMultipleGroupsJointLeader (0.01s)
=== RUN TestHeartbeatResponseFanout
I0524 01:27:44.746267      94 heartbeat_test.go:387] SubmitCommand succeed after Heartbeat Response fanout
E0524 01:27:44.746653      94 transport.go:202] sending rpc failed: read tcp 127.0.0.1:52991: use of closed network connection
E0524 01:27:44.746711      94 transport.go:202] sending rpc failed: read tcp 127.0.0.1:52991: use of closed network connection
E0524 01:27:44.746760      94 transport.go:202] sending rpc failed: read tcp 127.0.0.1:52991: use of closed network connection
E0524 01:27:44.746811      94 transport.go:202] sending rpc failed: read tcp 127.0.0.1:58890: use of closed network connection

```
Please assign, take a look and update the issue accordingly.

--- a/kv/dist_sender_server_test.go
+++ b/kv/dist_sender_server_test.go
@@ -183,11 +183,7 @@ func TestStartEqualsEndKeyScan(t *testing.T) {
 	}
 
 	call := client.Scan(proto.Key("a"), proto.Key("a"), 0)
-	sr := call.Reply.(*proto.ScanResponse)
-	if err := db.Run(call); err != nil {
-		t.Fatalf("unexpected error on scan: %s", err)
-	}
-	if l := len(sr.Rows); l != 0 {
-		t.Errorf("expected no rows; got %d", l)
+	if err := db.Run(call); err == nil {
+		t.Fatalf("expected error on scan with startkey == endkey")
 	}
 }

